protostar/commands/test_cairo1/test_cairo1_command.py

```diff
@@ -0,0 +1,139 @@
+        protostar_directory: ProtostarDirectory,
+        log_color_provider: LogColorProvider,
+        cwd: Path,
+        active_profile_name: Optional[str],
+        messenger_factory: MessengerFactory,
+    ) -> None:
+        super().__init__()
+        self._log_color_provider = log_color_provider
+        self._project_root_path = project_root_path
+        self._protostar_directory = protostar_directory
+        self._project_cairo_path_builder = LinkedLibrariesBuilder()
+        self._cwd = cwd
+        self._active_profile_name = active_profile_name
+        self._messenger_factory = messenger_factory
+
+    @property
+    def name(self) -> str:
+        return "test-cairo1"
+
+    @property
+    def description(self) -> str:
+        return "Executes cairo1 tests."
+
+    @property
+    def example(self) -> Optional[str]:
+        return "$ protostar test"
+
+    @property
+    def arguments(self):
+        return [
+            *MessengerFactory.OUTPUT_ARGUMENTS,
+            ProtostarArgument(
+                name="target",
+                description="""
+A glob or globs to a directory or a test suite, for example:
+- `tests/**/*_main*::*_balance` — find test cases, which names ends with `_balance` in test suites with the `_main` in filenames in the `tests` directory,
+- `::test_increase_balance` — find `test_increase_balance` test_cases in any test suite within the project.   
+""".strip(),
+                type="str",
+                value_parser="list",
+                is_positional=True,
+                default=["."],
+            ),
+            ProtostarArgument(
+                name="ignore",
+                short_name="i",
+                description=(
+                    "A glob or globs to a directory or a test suite, which should be ignored."
+                ),
+                value_parser="list",
+                type="str",
+            ),
+            ProtostarArgument(
+                name="linked-libraries",
+                value_parser="list",
+                description="Libraries to include in compilation",
+                type="path",
+            ),
+            ProtostarArgument(
+                name="no-progress-bar",
+                type="bool",
+                description="Disable progress bar.",
+            ),
+            ProtostarArgument(
+                name="exit-first",
+                short_name="x",
+                type="bool",
+                description="Exit immediately on first broken or failed test.",
+            ),
+            ProtostarArgument(
+                name="report-slowest-tests",
+                type="int",
+                description="Print slowest tests at the end.",
+                default=0,
+            ),
+            ProtostarArgument(
+                name="last-failed",
+                short_name="lf",
+                type="bool",
+                description="Only re-run failed and broken test cases.",
+            ),
```

Comment:
Do all of these work? We should add some tests for checking that

Reply:
Yup I know, there is a task for that 😉 
[#tests exten](https://github.com/software-mansion/protostar/issues/1544)